client/src/frontend/components/resource-slide/SlideInteractive.js

```diff
@@ -1,21 +1,125 @@
+
+export default class ResourceListSlideInteractive extends PureComponent {
+  static displayName = "ResourceList.Slide.Interactive";
 
-export default class ResourceListSlideInteractive extends Component {
   static propTypes = {
-    resource: PropTypes.object,
-    enableZoom: PropTypes.bool,
+    resource: PropTypes.object.isRequired,
+    flexibleHeight: PropTypes.bool,
     noPlaceholder: PropTypes.bool
   };
 
   static defaultProps = {
-    enableZoom: true,
-    zoomLabel: "View Interactive Resource",
+    flexibleHeight: false,
     noPlaceholder: false
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      containerWidth: 0,
+      containerHeight: 0,
+      poster: get(
+        this.resource,
+        "attributes.variantPosterStyles.mediumLandscape"
+      )
+    };
+
+    this.updateStateOnResize = throttle(this.updateContainerState, 100);
+  }
+
+  get resource() {
+    return this.props.resource;
+  }
+
+  componentDidMount() {
+    this.updateContainerState();
+    window.addEventListener("resize", this.updateStateOnResize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateStateOnResize);
+  }
+
+  setContainer = el => {
+    this.containerRef = el;
+  };
+
+  updateContainerState = () => {
+    this.setState({
+      containerWidth: this.containerWidth(),
+      containerHeight: this.containerHeight()
+    });
+  };
+
+  containerWidth() {
+    if (!this.containerRef) return 0;
+    return this.containerRef.offsetWidth;
+  }
+
+  containerHeight() {
+    if (!this.containerRef) return 0;
+    return this.containerRef.offsetHeight;
+  }
+
+  containerIsWideEnough() {
+    return this.minimumWidth() <= this.state.containerWidth;
+  }
+
+  containerIsTallEnough() {
+    if (this.props.flexibleHeight) return true;
+    return this.minimumHeight() <= this.state.containerHeight;
+  }
+
+  minimumHeight() {
+    const { minimumHeight } = this.props.resource.attributes;
+    if (!minimumHeight) return 800;
+    return minimumHeight;
+  }
+
+  minimumWidth() {
+    const { minimumWidth } = this.props.resource.attributes;
+    if (!minimumWidth) return 1020;
+    return minimumWidth;
+  }
+
+  canShowIFrame() {
+    if (this.state.poster) return false;
+    return this.containerIsWideEnough() && this.containerIsTallEnough();
+  }
+
   render() {
-    return <ResourcePlayer.Iframe {...this.props} />;
+    const { resource } = this.props;
+    return (
+      <figure ref={this.setContainer}>
+        {this.canShowIFrame() ? (
+          <>
+            <Zoom
+              zoom
+              label={"View interactive resource"}
+              resource={resource}
+            />
+            <ResourcePlayer.Iframe
+              resource={resource}
+              styleProps={{ height: "100%" }}
+            />
+          </>
+        ) : (
+          <Slide
+            resource={this.resource}
+            zoom
+            info={!this.state.poster}
```

Comment:
Unclear why poster should have anything to do with info. Methods would help me understand intent.